### PR TITLE
Fix missing artwork

### PR DESCRIPTION
**Pull request by michalmo**
Missing artwork has been reported at #1848 and https://github.com/home-assistant/core/issues/81185https://github.com/home-assistant/core/issues/81185.

My own investigation shows that this is not an issue isolated to HomePods. tvOS seems to only fetch the artwork locally for select media sources - it does for video media from streaming apps, but does not for music media from Apple Music (instead returns links to itunes artwork hosted by Apple).

This PR will fetch the artwork from the `artworkIdentifier` or `artworkURL` fields of the metadata if available.

Home Assistant supports returning artwork urls instead of artwork data, but that would require an API change in pyatv so I wend with this approach.

### CODE REVIEW

File: pyatv/protocols/mrp/__init__.py [hunk 1]
```diff
@@ -502,6 +505,58 @@ async def artwork(
...
+                url = None
+            url_parts = urlparse(url)
+            if url_parts.scheme and url_parts.netloc:
+                urls.append(url)
+
+        if metadata.HasField("artworkURL"):
+            # artworkURL has fixed size and format, use it as a fallback
+            urls.append(metadata.artworkURL)
+
+        async with aiohttp.ClientSession() as session:
```
Comment (postlund) [inline pyatv/protocols/mrp/__init__.py:541]: You should not create a new session here, but rather re-use the one in the `pyatv.core.Core` instance (passed when instantiating the protocol). That session is re-used everywhere in pyatv and is usually provided by the user.

**Review by postlund — CHANGES_REQUESTED** [review]
I have not tried it myself, but it looks solid to me. This is something I would like a test for though. It should be fairly simple to set up a small web server using `pyatv.support.http.BasicHttpServer`. Just provide a handler that returns a fixed value, make the fake MRP device include a URL to said web server and verify the fetch artwork matches what is returned by the handler.

**Review by postlund — APPROVED** [review]
Looks good! 👍

**Comment by haywiremk** [comment]
Any update on what version of HA(https://www.home-assistant.io/integrations/apple_tv) this fix may get merged into?

**Comment by michalmo** [comment]
There will need to be a new release that includes this PR, and then HA will be updated to point at this latest release. However it may take some more time, the project owner seems to be busy at the moment.

**Pull request merged**